test_scripts/test_main.py

```diff
@@ -567,7 +568,97 @@ def run_pytest(pargs, test_descr, test_dir, test_report,
     # We change the current directory so that the coverage output goes to the outputdir
     with subprocess.Popen(args,stderr=subprocess.STDOUT, stdout=subprocess.PIPE,cwd=pargs.outputdir) as p:
         while(p.poll() is None):
-            test_report.pytest_line(test_descr, p.stdout.readline().decode().strip())
+            # Scan the output for status
+            line = p.stdout.readline().decode().strip()
+            test_report.test_line(line)
+            if "warnings" in line or "passed" in line or "failed" in line:
+                # The output will look like
+                # ========== 47 passed, 29 warnings in 648.20s (0:10:48) =========
+                # Strip out the = signs
+                results =  line.replace("=", "")
+
+    test_report.add_test_section_summary(test_descr, "failed" not in results, results)
+
+    
+def is_git_installed():
+    """Detrmines if git is installed"""
+    git_path = shutil.which("git")
+    return git_path is not None
+
+def is_path_in_git(path):
+    """Determines if a given path is in a git repository"""
+    try:
+        result = subprocess.run(["git", "rev-parse", "--is-inside-work-tree"], stdout=subprocess.DEVNULL, cwd=path)
+    except Exception:
+        print(f"Failed to determine if {path} is in git.\n" + traceback.format_exc())
+        return False
+    return result.returncode == 0
+
+def run_update_docs(pargs, test_report):
+    """Run the PypeIT update_docs script and verify it succeeds.
+    
+    Args:
+        pargs (:obj:`argparse.Namespace`): The arguments to pypeit_test, as returned by argparse.
+
+        test_report (:obj:`TestReport`): 
+            The test report object for this test run. Output from the pytest run
+            will be passed to this object.
+
+    """
+    test_descr = "PypeIt Update Docs"
+    test_report.test_section_started(test_descr)
+
+    pypeit_dir = pypeit.__path__[0]
+    passed = False
+    # Default failure message if we can't find a status in the scripts output
+    status = red_text("Unknown Failure")
+
+
+    if is_git_installed() is False:
+        test_report.test_line("Skipping Update Docs because git is not installed")
+        status = yellow_text("SKIPPED: git not installed")
+        passed = True
+    elif is_path_in_git(pypeit_dir) is False:
+        test_report.test_line("Skipping Update Docs because PypeIt is not being run from git")
+        status = yellow_text("SKIPPED: PypeIt not in git")
+        passed = True
+    else:
```

Comment:
For my first point, I was referring to the code here.  I assumed (incorrectly?) that these checks for whether or not `git` is installed and that you're working in the `git` path had to do with the `git` commands in the `update_docs` script.  My point was that these `git` commands in `update_docs` aren't strictly necessary, so we might make these code changes simpler by removing them.